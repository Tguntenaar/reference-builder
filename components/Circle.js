import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

const Circle = ({ color, size }) => {
  return (
    <View
      style={{
        position: 'absolute',
        top: -size / 3,
        right: -size / 3,
        height: size,
        width: size,
        borderRadius: size / 2,
        backgroundColor: color,
        zIndex: -3,
      }}
    />
  );
};

Circle.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
};

Circle.defaultProps = {
  color: 'rgba(239, 244, 253, 0.2)',
  size: 300,
};

export default Circle;
