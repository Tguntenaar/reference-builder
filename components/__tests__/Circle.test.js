import React from 'react';
import renderer from 'react-test-renderer';
import Circle from '../Circle';

describe('<Circle />', () => {
  it('has 0 children', () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree.children).toBe(null);
  });
});

test('renders correctly', () => {
  const tree = renderer.create(<Circle />).toJSON();
  expect(tree).toMatchSnapshot();
});
