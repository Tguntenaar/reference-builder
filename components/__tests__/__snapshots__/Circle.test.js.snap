// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(239, 244, 253, 0.2)",
      "borderRadius": 150,
      "height": 300,
      "position": "absolute",
      "right": -100,
      "top": -100,
      "width": 300,
      "zIndex": -3,
    }
  }
/>
`;
